RealtimeJMSToStream/README.rst

```diff
@@ -0,0 +1,90 @@
+===============================================================================
+Realtime Java Message Service (JMS) Source To Stream Sink Adapter Configuration
+===============================================================================
+
+The ETL Realtime Template can be used to create an Adapter that reads from a Realtime Source and pass it to a Sink to be processed further.
+
+In this example, we will poll messages from Java Message Service (JMS) source in realtime. and use a Stream Sink to write the messages to CDAP Stream.
+
+The config.json contains a sample Adapter configuration that you can use to accomplish the above task. 
+
+Our sample Adapter uses these components:
+
+- ETLRealtime Application Template, since we want to perform ETL in realtime
+- JMS source, using the implementation of JMS from Apache ActiveMQ, version 5.11.1
+- Stream sink to hold the incoming messages to be processed by a downstream program such as a CDAP Flow
+
+You can create and start the Adapter by using the CDAP CLI (or you can use the UI for a more visual approach).
+
+The realtime JMS source uses generic JMS interface to communicate with JMS broker, so the source requires implementation of the JMS interfaces
+to be able to work properly as separate adapter plugin.
+
+One important thing to remember you need to use the identical version of JMS for both the client and the target server to ensure 
+that the message format is respected when the JMS source start polling messages.
+
+The realtime JMS source requires an additional plugin jar that bundle all JMS specific implementations into a jar. 
+For Apache ActiveMQ, we will be using the ``activemq-all-5.11.1.jar`` file that contains most of the required dependencies.
+
+For each additionl plugin jars, there should be accompanying JSON file with the same name as the jar name, but with .json extension, 
+to describe about the plugin so the adapter knows about it and load it as part of the plugon class loader. The jar and the JSON counterpart files
+should be put in the templates/plugins/ETLRealtime directory.
+
+There are additional jars that should be copied into templates/plugins/ETLRealtime/lib directory. These jars will be available as part of combined 
```

Comment:
This is because the way plugin class loader work. For special kind of plugin like JMS, which requires third party jars, it either get all the additional dependencies and its transitive dependencies in one jar file. This means that developers that uses JMS source for building adapter would need to build bundle jars that incorporate all dependencies.

But, some products like ActiveMQ has the "all" jar which contains most of the dependencies but leave some like JMS APIs to be available somewhere in the program class path.

In our case, we need to bring these API jars into shared directory (ie: templates/plugins/ETLRealtime/lib) so that JMS source class loader could get to those dependencies if the "all" jar does not contain one.

This is like a workaround we provide so that developers do not have to bundle all dependency jars when using Apache ActiveMQ.